components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/SyncEnginesStorage.kt

```diff
@@ -44,14 +49,31 @@ class SyncEnginesStorage(private val context: Context) {
      */
     fun setStatus(engine: SyncEngine, status: Boolean) {
         storage().edit().putBoolean(engine.nativeName, status).apply()
+        syncStore.publishChanges()
     }
 
     /**
      * Clears out any stored [SyncEngine] state.
      */
     internal fun clear() {
         storage().edit().clear().apply()
+        syncStore.publishChanges()
     }
 
     private fun storage() = context.getSharedPreferences(SYNC_ENGINES_KEY, Context.MODE_PRIVATE)
+
+    private fun SyncStore.publishChanges() {
+        val engines = getCurrentEngines().filter { it.value }.keys.toList()
+        dispatch(SyncAction.UpdateEnabledEngines(engines))
+    }
+
+    // TODO does this need to be reversed? Go through what we have in local storage, and populate
+    // result map based on that. reason: we may have "other" engines.
+    // this will be empty if `setStatus` was never called.
+    private fun getCurrentEngines(): Map<SyncEngine, Boolean> =
+        storage().all.map {
```

Comment:
I _think_ that I did this because there is nothing in the app that initially checks the status of the engines. This was a way to ensure that every enabled engine was added to the store any time the status of a single engine was checked or updated. Looking back at it though, there is probably a better way to do so.